scripts/CEE/collect-verbose-ca-logs/script.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+# Set variables
+GREEN="\033[32m"
+RESET="\033[0m"
+NAMESPACE="openshift-machine-api"
+CA_NAME="default"
+
+# Get current logVerbosity value
+current_log_verbosity=$(oc get ca $CA_NAME -o jsonpath='{.spec.logVerbosity}')
+
+# Check if current_log_verbosity is empty
+if [ -z "$current_log_verbosity" ]; then
+  echo "Failed to retrieve current logVerbosity value."
+  exit 1
+fi
+
+echo
+echo "CURRENT LOG VERBOSITY = $current_log_verbosity"
+echo
+
+# Update logVerbosity to 6
+oc patch ca $CA_NAME --type='json' -p='[{"op": "replace", "path": "/spec/logVerbosity", "value": 6}]'
+
+updated_log_verbosity=$(oc get ca $CA_NAME -o jsonpath='{.spec.logVerbosity}')
+echo
+echo "UPDATED LOG VERBOSITY = $updated_log_verbosity"
+echo
+
+# Wait for the update to take effect
+echo "Waiting for the log verbosity to update..."
+sleep 30
+
+# Get the name of the Cluster Autoscaler pod
+CA_POD=$(oc get pods -n openshift-machine-api | grep cluster-autoscaler-default | awk '{print $1}')
+
+# Collect logs for the next 6 minutes
+echo
+echo "Collecting logs for the next 6 minutes..."
+echo
+
+sleep 360
```

Comment:
In most of the cases, the CA checks the eligibility of a node for scale down if it is unneeded for 5 minutes OR 10 minutes. This is something that the customer can configure. Previously, I thought it was default to 5 but while working on a recent case I realize it could be 10 as well. TBH, I kept it 6 because in most of the case it worked for me. 

However, after some research I got to know about `--scale-down-unneeded-time=5m` arg which is defined in the CA Pod YAML. In my code, I will try to get this value and the script will collect the logs for **"scale-down-unneeded-time + 1" minutes.**

Reply:
Sorry, I just checked that the arg is only available if the customer configures it. I believe collecting it for 6  minutes should be fine as if we redeploy the pod, it will immediately start checking all the nodes. This will help us keep the script simple.

Reply:
thanks for explaining! that makes sense.